module/planning/PlanWalkPath/src/PlanWalkPath.cpp

```diff
@@ -74,15 +76,20 @@ namespace module::planning {
 
             // Turn on the spot
             emit<Task>(std::make_unique<Walk>(
-                Eigen::Vector3d(cfg.rotate_speed_x, cfg.rotate_speed_y, sign * cfg.rotate_speed)));
+                Eigen::Vector3d(cfg.rotate_velocity_x, cfg.rotate_velocity_y, sign * cfg.rotate_velocity)));
         });
 
         on<Provide<TurnAroundBall>>().then([this](const TurnAroundBall& turn_around_ball) {
             // Determine the direction of rotation
             int sign = turn_around_ball.clockwise ? -1 : 1;
             // Turn around the ball
-            emit<Task>(std::make_unique<Walk>(
-                Eigen::Vector3d(cfg.pivot_ball_speed_x, cfg.pivot_ball_speed_y, sign * cfg.pivot_ball_speed)));
+            emit<Task>(std::make_unique<Walk>(Eigen::Vector3d(cfg.pivot_ball_velocity_x,
+                                                              sign * cfg.pivot_ball_velocity_y,
+                                                              sign * cfg.pivot_ball_velocity)));
         });
+
+        // Reset the velocity to minimum velocity after a kick
+        on<Trigger<KickFinished>>().then(
+            [this](const KickFinished&) { velocity_magnitude = cfg.min_translational_velocity_magnitude; });
```

Comment:
we shouldn't have to do anything like this with Director, all modules should be self contained. Just check if there's no walk task, and if so then reset the velocity.